gems/sorbet-runtime/lib/types/props/optional.rb

```diff
@@ -21,12 +21,14 @@ module T::Props::Optional::DecoratorMethods
     true,
   ].freeze
 
-  def valid_props
-    super + [
-      :default,
-      :factory,
-      :optional,
-    ]
+  VALID_RULE_KEYS = Set[
```

Comment:
ditto, could we mark it private?

Reply:
Also, if you're looking for most perf, Hash is faster than Set

Reply:
Huh, that's silly. In a quick test looks like it's true specifically when you use `[]`, but `include?` is about the same either way, and maybe 50% slower - does that match your expectation? I guess `Hash#[]` has some special treatment?

Reply:
Made the constants private. I think I'll leave changing to Hash to when I'm able to delete `valid_props` entirely.